src/app/modules/group/http-services/search-group.service.ts

```diff
@@ -0,0 +1,36 @@
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { appConfig } from 'src/app/shared/helpers/config';
+
+export interface GroupFound<Types> {
+  id: string,
+  name: string,
+  description: string|null,
+  type: Types,
+}
+
+function notBase(group: GroupFound<'Class'|'Team'|'Club'|'Friends'|'Other'|'Base'>):
+  group is GroupFound<'Class'|'Team'|'Club'|'Friends'|'Other'> {
+  return group.type !== 'Base';
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SearchGroupService {
+
+  constructor(private http: HttpClient) { }
+
+  search(
```

Comment:
You overuse the generic types here. If you need two types with different allowed type, just define two types and maybe use "extends" if you do not want to repeat id, name, desc.

Reply:
fixed